packages/flutter/lib/src/material/search.dart

```diff
@@ -274,6 +280,13 @@ abstract class SearchDelegate<T> {
   /// [InputDecorationTheme.hintStyle] will be used instead.
   final TextStyle searchFieldStyle;
 
+  /// The [InputDecoration] for the search field, use
+  /// this if you just want to customize the [TextField]
+  ///
+  /// If this value is not null, [searchFieldStyle]
+  /// will be ignored so this can be used.
+  final InputDecorationTheme searchFieldDecoration;
```

Comment:
I would prefer naming this `searchFieldDecorationTheme` to avoid confusion with regular `InputDecoration` and `InputDecorator` classes.

Also, if we just ignore `searchFieldStyle` if this is set, perhaps we should have an assert in the constructor that ensures only one of these properties is set.